python/cuvs_bench/cuvs_bench/synthesize_dataset/_io.py

```
@@ -0,0 +1,170 @@
+    """
+    ext = os.path.splitext(path)[1].lower()
+
+    if ext == ".npy":
+        data = np.load(path)
+        if sample_size is not None and sample_size < len(data):
+            data = data[:sample_size]
+        return np.ascontiguousarray(data.astype(np.float32))
+
+    if ext == ".pkl":
+        with open(path, "rb") as f:
+            data = pickle.load(f)
```

Ruff (0.15.17)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)